socket_linux.cc: remove AI_V4MAPPED

--- a/dart/runtime/bin/socket_linux.cc
+++ b/dart/runtime/bin/socket_linux.cc
@@ -212,14 +212,14 @@ AddressList<SocketAddress>* Socket::LookupAddress(const char* host,
   memset(&hints, 0, sizeof(hints));
   hints.ai_family = SocketAddress::FromType(type);
   hints.ai_socktype = SOCK_STREAM;
-  hints.ai_flags = (AI_V4MAPPED | AI_ADDRCONFIG);
+  hints.ai_flags = AI_ADDRCONFIG;
   hints.ai_protocol = IPPROTO_TCP;
   struct addrinfo* info = NULL;
   int status = NO_RETRY_EXPECTED(getaddrinfo(host, 0, &hints, &info));
   if (status != 0) {
     // We failed, try without AI_ADDRCONFIG. This can happen when looking up
     // e.g. '::1', when there are no global IPv6 addresses.
-    hints.ai_flags = AI_V4MAPPED;
+    hints.ai_flags = 0;
     status = NO_RETRY_EXPECTED(getaddrinfo(host, 0, &hints, &info));
     if (status != 0) {
       ASSERT(*os_error == NULL);
