--- a/sdk/runtime/vm/globals.h	2015-10-01 13:47:09.000000000 -0700
+++ b/sdk/runtime/vm/globals.h	2016-04-11 15:40:07.013270085 -0700
@@ -27,8 +27,10 @@
 const intptr_t kSmiMax = (static_cast<intptr_t>(1) << kSmiBits) - 1;
 const intptr_t kSmiMin =  -(static_cast<intptr_t>(1) << kSmiBits);

-const double kPosInfinity = bit_cast<double>(DART_UINT64_C(0x7ff0000000000000));
-const double kNegInfinity = bit_cast<double>(DART_UINT64_C(0xfff0000000000000));
+const double kPosInfinity __attribute__((unused)) =
+    bit_cast<double>(DART_UINT64_C(0x7ff0000000000000));
+const double kNegInfinity __attribute__((unused)) =
+    bit_cast<double>(DART_UINT64_C(0xfff0000000000000));

 // The expression ARRAY_SIZE(array) is a compile-time constant of type
 // size_t which represents the number of elements of the given
