diff --git a/gcc/opts.c b/gcc/opts.c
index 6c8b3bc..d41a4a1 100644
--- a/gcc/opts.c
+++ b/gcc/opts.c
@@ -906,7 +906,7 @@ decode_options (unsigned int argc, const char **argv)
   flag_tree_pre = opt2;
   flag_tree_switch_conversion = opt2;
   flag_ipa_cp = opt2;
-  flag_ipa_sra = opt2;
+  flag_ipa_sra = 0 /* opt2 */;
 
   /* Track fields in field-sensitive alias analysis.  */
   set_param_value ("max-fields-for-field-sensitive",
