--- grpc-release-0_9_0.orig/Makefile	2015-05-26 18:04:29.000000000 -0700
+++ grpc-release-0_9_0/Makefile	2015-06-08 16:12:02.553192136 -0700
@@ -202,13 +202,12 @@
 LD = $(LD_$(CONFIG))
 LDXX = $(LDXX_$(CONFIG))
 AR = ar
+STRIP ?= strip
 ifeq ($(SYSTEM),Linux)
-STRIP = strip --strip-unneeded
+STRIP = $(STRIP) --strip-unneeded
 else
 ifeq ($(SYSTEM),Darwin)
-STRIP = strip -x
-else
-STRIP = strip
+STRIP = $(STRIP) -x
 endif
 endif
 INSTALL = install
