From a202c6b6efa0b3770ef674bd8a2dc7c6d813b531 Mon Sep 17 00:00:00 2001
From: Denton Gentry <dgentry@google.com>
Date: Fri, 19 Dec 2014 20:22:59 -0800
Subject: [PATCH] Set stdout to unbuffered, for better logging.

---
 src/dnsmasq.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/dnsmasq.c b/src/dnsmasq.c
index f4a89fc..e0f776a 100644
--- a/src/dnsmasq.c
+++ b/src/dnsmasq.c
@@ -81,6 +81,8 @@ int main (int argc, char **argv)
 
   umask(022); /* known umask, create leases and pid files as 0644 */
  
+  setvbuf(stdout, NULL, _IONBF, 0);
+
   rand_init(); /* Must precede read_opts() */
   
   read_opts(argc, argv, compile_opts);
-- 
2.2.0.rc0.207.ga3a616c

