From a0edaffb9bf21c241cc06440bd38ed727b86723a Mon Sep 17 00:00:00 2001
From: Will Angley <willangley@google.com>
Date: Wed, 22 Apr 2015 12:28:10 -0400
Subject: [PATCH 2/3] Remove iperf3_profile build.

---
 src/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index 665cd0a..f4e99e0 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -1,6 +1,6 @@
 lib_LTLIBRARIES         = libiperf.la                                   # Build and install an iperf library
 bin_PROGRAMS            = iperf3                                        # Build and install an iperf binary
-noinst_PROGRAMS         = t_timer t_units t_uuid iperf3_profile         # Build, but don't install the test programs and a profiled version of iperf3
+noinst_PROGRAMS         = t_timer t_units t_uuid
 include_HEADERS         = iperf_api.h # Defines the headers that get installed with the program
 
 
-- 
2.2.0.rc0.207.ga3a616c

