From 5e839aa2d9b7776f0bcd17964a0050d88461ade5 Mon Sep 17 00:00:00 2001
From: pzm <pzm@google.com>
Date: Tue, 27 Oct 2015 19:24:11 -0700
Subject: [PATCH] Patch gossl to support deadline.

---
 conn.go | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/conn.go b/conn.go
index a1437d7..91fd5a8 100644
--- a/conn.go
+++ b/conn.go
@@ -80,13 +80,13 @@ func (self *Conn) Write(b []byte) (int, error) {
 	return self.ssl.Write(b)
 }
 func (self *Conn) SetDeadline(t time.Time) error {
-	return nil
+	return self.conn.SetDeadline(t)
 }
 func (self *Conn) SetReadDeadline(t time.Time) error {
-	return nil
+	return self.conn.SetReadDeadline(t)
 }
 func (self *Conn) SetWriteDeadline(t time.Time) error {
-	return nil
+	return self.conn.SetWriteDeadline(t)
 }
 func (self *Conn) Handshake() (err error) {
 	if !self.handshakeCompleted {
-- 
2.6.0.rc2.230.g3dd15c0

