From 24af2fa43747fae111669c33a7877f0ab85b5cd0 Mon Sep 17 00:00:00 2001
From: Avery Pennarun <apenwarr@gmail.com>
Date: Fri, 4 Mar 2016 23:46:22 -0500
Subject: [PATCH] Provide unshare() if missing from libc.

---
 configure    | 24 ++++++++++++++++++++++++
 ip/Makefile  |  4 ++++
 ip/ipnetns.c | 12 ++++++++++++
 3 files changed, 40 insertions(+)

diff --git a/configure b/configure
index d5170f0..e3e0cc4 100755
--- a/configure
+++ b/configure
@@ -200,6 +200,27 @@ EOF
     rm -f $TMPDIR/setnstest.c $TMPDIR/setnstest
 }
 
+check_unshare()
+{
+    cat >$TMPDIR/unsharetest.c <<EOF
+#include <sched.h>
+int main(int argc, char **argv)
+{
+	(void)unshare(0);
+	return 0;
+}
+EOF
+    $CC -I$INCLUDE -o $TMPDIR/setnstest $TMPDIR/unsharetest.c >/dev/null 2>&1
+    if [ $? -eq 0 ]
+    then
+	echo "IP_CONFIG_UNSHARE:=y" >>Config
+	echo "yes"
+    else
+	echo "no"
+    fi
+    rm -f $TMPDIR/unsharetest.c $TMPDIR/unsharetest
+}
+
 check_ipset()
 {
     cat >$TMPDIR/ipsettest.c <<EOF
@@ -266,5 +287,8 @@ check_ipt_lib_dir
 echo -n "libc has setns: "
 check_setns
 
+echo -n "libc has unshare: "
+check_unshare
+
 echo -n "SELinux support: "
 check_selinux
diff --git a/ip/Makefile b/ip/Makefile
index 713adf5..840132e 100644
--- a/ip/Makefile
+++ b/ip/Makefile
@@ -15,6 +15,10 @@ ifeq ($(IP_CONFIG_SETNS),y)
 	CFLAGS += -DHAVE_SETNS
 endif
 
+ifeq ($(IP_CONFIG_UNSHARE),y)
+	CFLAGS += -DHAVE_UNSHARE
+endif
+
 ALLOBJ=$(IPOBJ) $(RTMONOBJ)
 SCRIPTS=ifcfg rtpr routel routef
 TARGETS=ip rtmon
diff --git a/ip/ipnetns.c b/ip/ipnetns.c
index 633b5b9..b08e23c 100644
--- a/ip/ipnetns.c
+++ b/ip/ipnetns.c
@@ -54,6 +54,18 @@ static int setns(int fd, int nstype)
 }
 #endif /* HAVE_SETNS */
 
+#ifndef HAVE_UNSHARE
+static int unshare(int flags)
+{
+#ifdef __NR_unshare
+	return syscall(__NR_unshare, flags);
+#else
+	errno = ENOSYS;
+	return -1;
+#endif
+}
+#endif /* HAVE_UNSHARE */
+
 static int usage(void)
 {
 	fprintf(stderr, "Usage: ip netns list\n");
-- 
2.7.0.rc3.207.g0ac5344

