--- a/fdisk/fdiskbsdlabel.h	2016-09-01 12:06:32.555493245 -0700
+++ b/fdisk/fdiskbsdlabel.h	2016-09-01 12:07:48.688709127 -0700
@@ -48,7 +48,8 @@
 
 #if defined (i386) || defined (__sparc__) || defined (__arm__) || \
     defined (__mips__) || defined (__s390__) || defined (__sh__) || \
-    defined(__x86_64__) || defined (__avr32__) || defined(__cris__)
+    defined(__x86_64__) || defined (__avr32__) || defined(__cris__) || \
+    defined(__arc__)
 #define BSD_LABELSECTOR   1
 #define BSD_LABELOFFSET   0
 #elif defined (__alpha__) || defined (__powerpc__) || defined (__ia64__) || defined (__hppa__)
