diff --git a/client.go b/client.go
index 670f4d4..d782cc3 100644
--- a/client.go
+++ b/client.go
@@ -8,9 +8,9 @@ import (
 	"sync"
 	"time"

-	"github.com/hashicorp/go.net/ipv4"
-	"github.com/hashicorp/go.net/ipv6"
 	"github.com/miekg/dns"
+	"golang.org/x/net/ipv4"
+	"golang.org/x/net/ipv6"
 )

 // ServiceEntry is returned after we query for a service
--- a/server.go
+++ b/server.go
@@ -6,9 +6,9 @@
 	"net"
 	"sync"

-	"github.com/hashicorp/go.net/ipv4"
-	"github.com/hashicorp/go.net/ipv6"
 	"github.com/miekg/dns"
+	"golang.org/x/net/ipv4"
+	"golang.org/x/net/ipv6"
 )

 const (
