--- a/net/bluetooth/smp.c	2016-01-12 08:34:03.114864045 -0800
+++ b/net/bluetooth/smp.c	2016-01-12 08:35:23.422411223 -0800
@@ -668,10 +668,12 @@ static void build_pairing_cmd(struct l2c
 		struct oob_data *oob_data;
 		u8 bdaddr_type;
 
+#ifndef CONFIG_BRUNO	/* GFRM200 rejects these bits */
 		if (hci_dev_test_flag(hdev, HCI_SSP_ENABLED)) {
 			local_dist |= SMP_DIST_LINK_KEY;
 			remote_dist |= SMP_DIST_LINK_KEY;
 		}
+#endif
 
 		if (hcon->dst_type == ADDR_LE_DEV_PUBLIC)
 			bdaddr_type = BDADDR_LE_PUBLIC;
